src/library/clementine/clementineplaylistmodel.cpp

```diff
@@ -0,0 +1,298 @@
+#include "library/clementine/clementineplaylistmodel.h"
+
+#include <QtDebug>
+
+#include "library/dao/playlistdao.h"
+#include "library/queryutil.h"
+#include "library/starrating.h"
+#include "library/trackcollectionmanager.h"
+#include "library/trackmodel.h"
+#include "mixer/playermanager.h"
+#include "track/beatfactory.h"
+#include "track/beats.h"
+#include "track/track.h"
+
+#define Clementine_TABLE "Clementine"
+#define CLM_VIEW_ORDER "position"
+#define CLM_ARTIST "artist"
+#define CLM_TITLE "title"
+#define CLM_DURATION "duration"
+#define CLM_URI "location"
+#define CLM_ALBUM "album"
+#define CLM_ALBUM_ARTIST "album_artist"
+#define CLM_YEAR "year"
+#define CLM_RATING "rating"
+#define CLM_GENRE "genre"
+#define CLM_GROUPING "grouping"
+#define CLM_TRACKNUMBER "tracknumber"
+#define CLM_DATEADDED "datetime_added"
+#define CLM_BPM "bpm"
+#define CLM_BITRATE "bitrate"
+#define CLM_COMMENT "comment"
+#define CLM_PLAYCOUNT "timesplayed"
+#define CLM_COMPOSER "composer"
+#define CLM_PREVIEW "preview"
+
+ClementinePlaylistModel::ClementinePlaylistModel(QObject* pParent,
+        TrackCollectionManager* pTrackCollectionManager,
+        ClementineDbConnection* pConnection)
+        : BaseSqlTableModel(pParent,
+                  pTrackCollectionManager,
+                  "mixxx.db.model.Clementine_playlist"),
+          m_pConnection(pConnection),
+          m_playlistId(-1) {
+}
+
+void ClementinePlaylistModel::setTableModel(int playlistId) {
+    qDebug() << "ClementinePlaylistModel::setTableModel" << playlistId;
+    if (m_playlistId == playlistId) {
+        qDebug() << "Already focused on playlist " << playlistId;
+        return;
+    }
+
+    if (m_playlistId >= 0) {
+        // Clear old playlist
+        m_playlistId = -1;
+        QSqlQuery query(m_database);
+        QString strQuery("DELETE FROM " Clementine_TABLE);
+        if (!query.exec(strQuery)) {
+            LOG_FAILED_QUERY(query);
+        }
+    }
+
+    if (playlistId >= 0) {
+        // setup new playlist
+        m_playlistId = playlistId;
+
+        QSqlQuery query(m_database);
+        QString strQuery(
+                "CREATE TEMP TABLE IF NOT EXISTS " Clementine_TABLE
+                " (" CLM_VIEW_ORDER " INTEGER, " CLM_ARTIST " TEXT, " CLM_TITLE
+                " TEXT, " CLM_DURATION " INTEGER, " CLM_URI " TEXT, " CLM_ALBUM
+                " TEXT, " CLM_ALBUM_ARTIST " TEXT, " CLM_YEAR
+                " INTEGER, " CLM_RATING " INTEGER, " CLM_GENRE
+                " TEXT, " CLM_GROUPING " TEXT, " CLM_TRACKNUMBER " INTEGER, "
+                //CLM_DATEADDED " INTEGER, "
```

Comment:
Why commented out? Prefer to delete unused code. Otherwise provide a comment that explains why this piece of code needs to be preserved as a comment.